### ISSUE TITLE
Refactor test_errors.py

### ISSUE THREAD

**Issue by Jasha10** [omry/omegaconf#537]
test_errors should probably be reorganized and broken into smaller functions (while reusing the error validation logic).

_Originally posted by @omry in https://github.com/omry/omegaconf/pull/503#discussion_r570537028_

**Comment by omry** [comment]
Please wait with this until we merge #445,

**Comment by Jasha10** [comment]
What would be a good first step?
Break the list of test cases into several sub-lists?

**Comment by omry** [comment]
Yeah, potentially calling a common function from various topic specific files.

Keep in mind that for now we have a 2.1 release branch and bug fixes in master are still likely to be ported to that branch. for that reason I think we should wait a while and only do this after Hydra 1.1 is released and had some time to stabilize (otherwise we will need to port the refactoring to the 2.1_branch to avoid conflicts when we cherry-pick other fixes to it).
